### PR TITLE
Notifications

### PR DESCRIPTION
**Pull request by ibrechin**
Basic backend support for notifications in the intelligence tool.

### CODE REVIEW

**Comment by ibrechin** [comment]
Obviously there's still more stuff that needs to be done, but I think that this is a fairly self-contained chunk to start with.

File: mtp_api/apps/notification/app.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from django.apps import AppConfig as DjangoAppConfig
+from django.utils.translation import gettext_lazy as _
+
+
+class AppConfig(DjangoAppConfig):
```
Comment (ushkarev) [inline mtp_api/apps/notification/app.py]: you need `default_app_config = 'notification.app.AppConfig'` in __init__.py

File: mtp_api/apps/notification/rules.py [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+from security.models import PrisonerProfile, SenderProfile, RecipientProfile
+from .models import Event, EventCredit, EventDisbursement
+
+INPUT_TYPES = Choices(
+    ('NUMBER', 'number', _('Number')),
+    ('STRING', 'string', _('String')),
+)
+
+
+class BaseRule():
```
Comment (ushkarev) [inline mtp_api/apps/notification/rules.py]: does this stop it being a subclass of object? probs best without `()`

Comment (ibrechin) [reply mtp_api/apps/notification/rules.py]: No, but sure.

File: mtp_api/apps/notification/rules.py [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+        [
+            time_period_input,
+            Input('totals__recipient_count__gte', INPUT_TYPES.NUMBER),
+        ],
+        profile=PrisonerProfile
+    ),
+    'CPNUM': TimePeriodQuantityRule(
+        Credit,
+        _(
+            'A debit card of bank account sending money to more than '
```
Comment (ushkarev) [inline mtp_api/apps/notification/rules.py]: … debit card _or_ bank account …

**Pull request closed without merging**